lib/spack/spack/solver/asp.py

```diff
@@ -1878,12 +1879,12 @@ def external_spec_selected(self, pkg, idx):
         )
 
     def depends_on(self, pkg, dep, type):
-        dependency = self._specs[pkg]._dependencies.get(dep)
-        if not dependency:
-            self._specs[pkg]._add_dependency(
-                self._specs[dep], (type,))
+        dependencies = self._specs[pkg].edges_to_dependencies(name=dep)
+        if not dependencies:
+            self._specs[pkg].add_dependency_edge(self._specs[dep], (type,))
```

Comment:
I think this interface ends up being confusing for dependencies.  There will still *never* be two of these.  I think we should avoid using a symmetric interface here -- `edges_to_dependencies()` makes sense to me but `edges_to_dependencies(name)` introduces a bunch of extra logic that we usually do not need.  Can we add `edge_to_dependency(name)` that returns only one thing, to simplify?

A singular `edge_from_dependent()` routine should probably *not* exist, as you could have multiple of them, and I think the client code should have to think about that.

`dependencies` just throws off my thinking here because there will never be more than one of these *for a given node*.

Reply:
See my comment below, but as discussed on our meeting today this is incorrect for future work planned.

Reply:
resolving per discussion points below.